### PR TITLE
ALTAPPS-576: Kotlin 1.8.21

### PR DESCRIPTION
**Pull request by XanderZhu**
**YouTrack Issues**:
[#ALTAPPS-576](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-576)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**

1. Update Kotlin & serialization to version 1.8.22
2. Update Ktor to version 2.3.3
3. Update Coroutines to version 1.7.2
4. Update Compose libraries

### CODE REVIEW

**Comment by XanderZhu** [comment]
@ivan-magda 
There is a compilation error for iOS build.
The reason for it is the [Ktor issue](https://youtrack.jetbrains.com/issue/KTOR-5728). We should wait until Ktor releases a new version with a new coroutines version.

**Comment by ivan-magda** [comment]
> @ivan-magda There is a compilation error for iOS build. The reason for it is the [Ktor issue](https://youtrack.jetbrains.com/issue/KTOR-5728). This PR should not be merged until Ktor releases a new version with a new Coroutine version.

@XanderZhu just checked out that the iOS app successfully compiling for release build after updating to ktor 2.3.3

File: gradle/libs.versions.toml
Comment (ivan-magda) [inline gradle/libs.versions.toml:3]: A newer version of Kotlin is available 1.8.22

File: gradle/libs.versions.toml
Comment (ivan-magda) [inline gradle/libs.versions.toml:4]: A newer version of coroutines is available: 1.7.3
https://github.com/Kotlin/kotlinx.coroutines/releases

File: gradle/libs.versions.toml
Comment (ivan-magda) [inline gradle/libs.versions.toml:18]: A newer version of androidx.compose:compose-bom than 2023.05.01 is available: 2023.08.00

**Review by ivan-magda — APPROVED** [review]
approved these changes to not block the merging

@XanderZhu before merging this PR into develop please delete all caches in GitHub Actions for develop branch, this is necessary to resolve possible problems with caches (a lot of dependencies changed)

https://github.com/hyperskill/mobile-app/actions/caches?query=branch%3Adevelop

File: gradle/libs.versions.toml
Comment (XanderZhu) [reply gradle/libs.versions.toml]: I update kotlinCompilerExtension and composeBom to the compatible with Kotlin 1.8.22 versions

**Comment by XanderZhu** [comment]
This PR should be merged when the 1.29 version will be released

**Pull request merged**
